cmd/minikube/cmd/config/addons_list.go

```diff
@@ -104,24 +100,43 @@ var printAddonsList = func(cc *config.ClusterConfig) {
 	table.SetAutoFormatHeaders(true)
 	table.SetBorders(tablewriter.Border{Left: true, Top: true, Right: true, Bottom: true})
 	table.SetCenterSeparator("|")
+
+	var tHeader []string
 	if cc == nil {
-		table.SetHeader([]string{"Addon Name", "Maintainer"})
+		tHeader = []string{"Addon Name", "Maintainer"}
 	} else {
-		table.SetHeader([]string{"Addon Name", "Profile", "Status", "Maintainer"})
+		tHeader = []string{"Addon Name", "Profile", "Status", "Maintainer"}
+	}
+	if printDocs {
+		tHeader = append(tHeader, "Docs")
 	}
+	table.SetHeader(tHeader)
 
 	for _, addonName := range addonNames {
 		addonBundle := assets.Addons[addonName]
 		maintainer := addonBundle.Maintainer
 		if maintainer == "" {
-			maintainer = "unknown (third-party)"
+			maintainer = "3rd party (unknown)"
+		}
+		docs := addonBundle.Docs
+		if docs == "" {
+			docs = "n/a"
 		}
 		if cc == nil {
-			tData = append(tData, []string{addonName, maintainer})
+			if printDocs {
+				tData = append(tData, []string{addonName, maintainer, docs})
+			} else {
+				tData = append(tData, []string{addonName, maintainer})
+			}
 			continue
+		} else {
+			enabled := addonBundle.IsEnabled(cc)
+			if printDocs {
+				tData = append(tData, []string{addonName, cc.Name, fmt.Sprintf("%s %s", stringFromStatus(enabled), iconFromStatus(enabled)), maintainer, docs})
+			} else {
+				tData = append(tData, []string{addonName, cc.Name, fmt.Sprintf("%s %s", stringFromStatus(enabled), iconFromStatus(enabled)), maintainer})
```

Comment:
These lines are decently long and mostly repetition, what do you think about doing something like the following?
```
tData = append(tData, []string{addonName, cc.Name, fmt.Sprintf("%s %s", stringFromStatus(enabled), iconFromStatus(enabled)), maintainer})
if printDocs {
    tData = append(tData, docs)
}
```

Could also do to the if case above.

Reply:
I tried your suggestion but it did not work out for me, since the first if statement `continue`s. That's because the first if statement is not checking the enablement of each addon. I could get rid of the `continue` but this would add runtime and checks, even though we already know that there's no configuration context.

Reply:
I was just referencing within the `else` block itself
```
} else {
    enabled := addonBundle.IsEnabled(cc)
    tData = append(tData, []string{addonName, cc.Name, fmt.Sprintf("%s %s", stringFromStatus(enabled), iconFromStatus(enabled)), maintainer})
    if printDocs {
        tData = append(tData, docs)
    }
}
```
I could be wrong, but I think you're referencing something else.

Reply:
This is not working. I have to append to each row of a 2d slice. If I'm doing it like this I'll append to the first element, e.g.:

```
$ ./out/minikube-darwin-amd64 addons list --docs
|-----------------------------------------------------------------------------------|--------------------------------|------|
|                                    ADDON NAME                                     |           MAINTAINER           | DOCS |
|-----------------------------------------------------------------------------------|--------------------------------|------|
| ambassador                                                                        | 3rd party (Ambassador)         |
| https://minikube.sigs.k8s.io/docs/tutorials/ambassador_ingress_controller/        |
| auto-pause                                                                        | Google                         |
| n/a                                                                               |
```

Reply:
I tried, but I just don't know how to extend each row of a 2d slice with a new column. All solutions I found are way more complicated than the current solution. Using temporary 2d slices, loops and copies...

Do you have another solution? Otherwise, I propose to keep it like it is, even though the line is repeated.